spec/models/classification_spec.rb

```diff
@@ -61,14 +61,55 @@
       FactoryGirl.create(:classification_tag,      :name => "multi_entry_2", :parent => parent)
     end
 
-    it "#destroy" do
-      cat = Classification.find_by_name("test_category")
-      expect(cat).to_not be_nil
-      entries = cat.entries
-      expect(entries.length).to eq(2)
+    context "#destroy" do
+      it "a category deletes all entries" do
+        cat = Classification.find_by_name("test_category")
+        expect(cat).to_not be_nil
+        entries = cat.entries
+        expect(entries.length).to eq(2)
+
+        cat.destroy
+        entries.each { |e| expect(Classification.find_by(:id => e.id)).to be_nil }
+      end
+
+      it "a category deletes assignments referenced by its entries" do
+        cat = Classification.find_by_name("test_category")
+        assignment_tag = "/chargeback_rate/assigned_to/vm/tag/managed/test_category/test_entry"
+
+        Tag.create!(:name => assignment_tag)
+        expect(Tag.exists?(:name => assignment_tag)).to be true
+
+        cat.destroy
+
+        expect(Tag.exists?(:name => assignment_tag)).to be false
+      end
 
-      cat.destroy
-      entries.each { |e| expect(Classification.find_by(:id => e.id)).to be_nil }
+      it "a category does not delete assignments that are close but do not match its tag" do
+        cat = Classification.find_by_name("test_category")
+        assignment_tag = "/chargeback_rate/assigned_to/vm/tag/managed/test_category/test_entry"
+        another_tag1 = Tag.create(:name => "/policy_set/assigned_to/vm/tag/managed/test_category/test_entry1")
+        another_tag2 = Tag.create(:name => "/chargeback_rate/assigned_to/vm/tag/managed/test_category1/test_entry")
+
+        Tag.create!(:name => assignment_tag)
+
+        cat.destroy
+
```

Comment:
We probably want this just to show that it deletes the correct tag but not the others (on the next two lines)
`expect(Tag.exists?(:name => assignment_tag)).to be false`